arch/sim/src/sim/up_macho_init.c

```diff
@@ -0,0 +1,100 @@
+/****************************************************************************
+ * arch/sim/src/sim/up_macho_init.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <assert.h>
+#include <stdlib.h>
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+typedef void (*init_func_t)(int argc, const char *argv[],
+                            const char *envp[], const char *apple[]);
+extern init_func_t mod_init_func_start \
+__asm("section$start$__DATA$__mod_init_func");
```

Comment:
can you reuse Linux approach to fix the early constuctor issue?

Reply:
i have tried but i couldn't. unlike linux, the dyld looks at section flags, not names. there seems no good tool to alter the flags.

Reply:
can we change the section flag too?

Reply:
can you explain?

Reply:
From what I understand is that:
1. rename the name of constructor section to _sinit
2. remove the macOS special flag from constructor section
then, macOS c/c++ runtime shouldn't find the constructor section anymore, and instead NuttX initialize code can find this section now.

Reply:
as i said above, there is no convenient way to remove "the macOS special flag" as far as i know.

Reply:
If so, how can you stop the compiler invoke c++ constructor before entering main?

Reply:
i dunno. i wrote this code because i couldn't find any nicer way.